utility.py

```diff
@@ -38,6 +39,34 @@ def distance(pos1: Tuple[float, float], pos2: Tuple[float, float]):
     return np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
 
 
+def k_means_clusters(net: sumolib.net.Net, num_clusters: int):
+    from perlin import get_edge_pair_centroid
+    edges = net.getEdges()
+
+    centroid_edges = [get_edge_pair_centroid(edge.getShape()) for edge in edges]
+    centroids = kmeans(centroid_edges, num_clusters, iter=25)
+
+    clusters = [[] for x in range(num_clusters)]
```

Comment:
Very minor thing, but I bet Pycharm is mad that `x` is unused - you can rename it to `_` and it'll shut up.